tests/unit/test-cmd/test.lint.js

```diff
@@ -65,6 +65,14 @@ describe('lint', () => {
     });
   });
 
+  it('passes warningsAsErrors to the linter', () => {
+    const {lint, createLinter} = setUp();
+    return lint({warningsAsErrors: true}).then(() => {
+      const config = createLinter.firstCall.args[0].config;
+      assert.equal(config.warningsAsErrors, true);
+    });
+  });
```

Comment:
this looks great! I think it would make sense to add one more test that does `lint()` without specifying the warnings flag and asserts the linter is called with `warningsAsErrors = undefined` (or should it be false?)